### PR TITLE
Move studios image providers to plugin

### PR DESCRIPTION
**Pull request by MrTimscampi**
**Changes**

Moves the built-in studio artwork provider to a plugin and allow customizing the repository URL.

This is the first step towards making a Jellyfin-owned artwork repository for studios and other similar things, as well as allowing users to change the URL if they want.

**Issues**

### CODE REVIEW

File: MediaBrowser.Providers/Plugins/StudiosImages/StudiosImageProvider.cs [hunk 1]
```diff
@@ -104,19 +107,19 @@ private RemoteImageInfo GetImage(BaseItem item, string filename, ImageType type,
...
         private string GetUrl(string image, string filename)
         {
-            return string.Format(CultureInfo.InvariantCulture, "https://raw.github.com/MediaBrowser/MediaBrowser.Resources/master/images/imagesbyname/studios/{0}/{1}.jpg", image, filename);
+            return string.Format(CultureInfo.InvariantCulture, "{0}/{1}/{2}.jpg", repositoryUrl, image, filename);
         }
 
         private Task<string> EnsureThumbsList(string file, CancellationToken cancellationToken)
         {
-            const string url = "https://raw.github.com/MediaBrowser/MediaBrowser.Resources/master/images/imagesbyname/studiothumbs.txt";
+            string url = string.Format("{0}/studiothumbs.txt", repositoryUrl);
```
Comment (crobibero) [inline MediaBrowser.Providers/Plugins/StudiosImages/StudiosImageProvider.cs:115]:
```suggestion
            string url = string.Format(CultureInfo.InvariantCulture, "{0}/studiothumbs.txt", repositoryUrl);
```

File: MediaBrowser.Providers/Plugins/StudiosImages/StudiosImageProvider.cs [hunk 2]
```diff
@@ -104,19 +107,19 @@ private RemoteImageInfo GetImage(BaseItem item, string filename, ImageType type,
...
 
             return EnsureList(url, file, _fileSystem, cancellationToken);
         }
 
         private Task<string> EnsurePosterList(string file, CancellationToken cancellationToken)
         {
-            const string url = "https://raw.github.com/MediaBrowser/MediaBrowser.Resources/master/images/imagesbyname/studioposters.txt";
+            string url = string.Format("{0}/studioposters.txt", repositoryUrl);
```
Comment (crobibero) [inline MediaBrowser.Providers/Plugins/StudiosImages/StudiosImageProvider.cs:122]:
```suggestion
            string url = string.Format(CultureInfo.InvariantCulture, "{0}/studioposters.txt", repositoryUrl);
```

File: MediaBrowser.Providers/Plugins/StudiosImages/Plugin.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            : base(applicationPaths, xmlSerializer)
+        {
+            Instance = this;
+        }
+
+        public static Plugin Instance { get; private set; }
+
+        public override Guid Id => new Guid("872a7849-1171-458d-a6fb-3de3d442ad30");
+
+        public override string Name => "StudiosImages";
```
Comment (crobibero) [inline MediaBrowser.Providers/Plugins/StudiosImages/Plugin.cs:24]:
```suggestion
        public override string Name => "Studio Images";
```

File: MediaBrowser.Providers/Plugins/StudiosImages/Plugin.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+        public override string Description => "Get artwork for studios from any Jellyfin-compatible repository.";
+
+        // TODO change this for a Jellyfin-hosted repository.
+        public const string DefaultServer = "https://raw.github.com/MediaBrowser/MediaBrowser.Resources/master/images/imagesbyname";
+
+        // TODO remove when plugin removed from server.
+        public override string ConfigurationFileName => "Jellyfin.Plugin.StudiosImages.xml";
```
Comment (crobibero) [inline MediaBrowser.Providers/Plugins/StudiosImages/Plugin.cs:32]:
```suggestion
        public override string ConfigurationFileName => "Jellyfin.Plugin.StudioImages.xml";
```

File: MediaBrowser.Providers/Plugins/StudiosImages/Configuration/config.html [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+                        Dashboard.hideLoadingMsg();
+                    });
+                });
+
+            document.querySelector('.configForm')
+                .addEventListener('submit', function (e) {
+                    Dashboard.showLoadingMsg();
+
+                    ApiClient.getPluginConfiguration(PluginConfig.pluginId).then(function (config) {
```
Comment (crobibero) [inline MediaBrowser.Providers/Plugins/StudiosImages/Configuration/config.html:46]:
```suggestion
                    window.ApiClient.getPluginConfiguration(PluginConfig.pluginId).then(function (config) {
```

iirc `ApiClient` is now a part of `window`

Comment (MrTimscampi) [reply MediaBrowser.Providers/Plugins/StudiosImages/Configuration/config.html:46]: Hm, we might have to update MusicBrainz and AudioDB then, this was taken from there.

**Review by BaronGreenback — APPROVED** [review]

File: MediaBrowser.Providers/Plugins/StudioImages/Configuration/config.html [hunk 6]
```diff
@@ -0,0 +1,58 @@
+﻿<!DOCTYPE html>
+<html>
+<head>
+    <title>AudioDB</title>
```
Comment (Ullmie02) [inline MediaBrowser.Providers/Plugins/StudioImages/Configuration/config.html:4]: Copy/Paste :)

**Review by crobibero — APPROVED** [review]

**Pull request merged**